src/model/channel/mod.rs

```diff
@@ -322,10 +265,7 @@ impl<'de> Deserialize<'de> for Channel {
             1 => serde_json::from_value::<PrivateChannel>(Value::Object(v))
                 .map(|x| Channel::Private(Arc::new(RwLock::new(x))))
                 .map_err(DeError::custom),
-            3 => serde_json::from_value::<Group>(Value::Object(v))
-                .map(|x| Channel::Group(Arc::new(RwLock::new(x))))
-                .map_err(DeError::custom),
-            4 => serde_json::from_value::<ChannelCategory>(Value::Object(v))
+            3 => serde_json::from_value::<ChannelCategory>(Value::Object(v))
```

Comment:
Category channels still have the ID 4, not 3, and since this PR was merged, deserializing a category into a Channel enum will throw "Unknown channel type".

Reply:
[Change reverted](https://github.com/serenity-rs/serenity/commit/5e5f0f9ca0592ada9df09f9ba3a6dc0121aba044).
This was a silly mistake of mine. I managed to notice the same attempt at disregarding the existence of `ChannelType` *4* at https://github.com/serenity-rs/serenity/pull/801#discussion_r389074242 but failed here. :stuck_out_tongue: